### PR TITLE
Enable Quadratic Elements for Patterned Peripheral Modifiers

### LINKED ISSUE #28296: Quadratic Element Capabilities for PatternedHex/CartesianPeripheralModifier
### Motivation

Original Reactor Module tools are only capable of creating meshes with linear elements, which limits its applications.

Efforts have been made to enable quadratic element capabilities. This enhancement follows up on #26986 #28164 #28169

### Design

PatternedHex/CartesianPeripheralModifier will automatically detect the order of input mesh and do the modification using the proper element type (TRI3 or TRI6).

### Impact

Quadratic element capability would benefit users with related needs and thus enhance the Reactor Module as well as MOOSE's intrinsic meshing capabilities.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #28296

### CODE REVIEW

File: modules/reactor/src/meshgenerators/PatternedPolygonPeripheralModifierBase.C [hunk 1]
```diff
@@ -235,6 +269,10 @@ PatternedPolygonPeripheralModifierBase::generate()
                                                                    _type,
                                                                    _name);
 
+    // Convert the linear mesh to quadratic mesh
+    if (order == 2)
+      mesh->all_second_order();
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PatternedPolygonPeripheralModifierBase.C:274]: let's keep track of quad8 or 9, then use all_second_order for quad8 and all_complete_order (if it exists) for quad9

**Review by GiudGiud — CHANGES_REQUESTED** [review]
looks good just this one comment

**Comment by GiudGiud** [comment]
so in that last commit you re-gold the file that is added in the first commit. let's try to squash that change into the 1st one so we don't make the repo grow

**Comment by miaoyinb** [comment]
Done

**Review by GiudGiud — APPROVED** [review]

**Comment by miaoyinb** [comment]
@GiudGiud Any idea why the module parallel test is cancelled by itself?

**Comment by GiudGiud** [comment]
No idea. I think there are mass cancellations buttons in civet that are used during maintenance/ to free up resources

**Pull request merged**